scipy/spatial/transform/rotation.py

```diff
@@ -1639,3 +1639,263 @@ def __call__(self, times):
+        Tolerance with which to calculate intermediate angular velocities.
+        Iterations are stopped when further steps change and magnitude of the
+        velocities by less than this amount. Default is 1e-15.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Spline
+            <http://qspline.sourceforge.net/qspline.pdf>`_
+    """
+    def __init__(self, times, rotations, omega_i=[0, 0, 0], omega_f=[0, 0, 0],
+                 tol=1e-12):
+        if len(rotations) == 1:
+            raise ValueError("`rotations` must contain at least 2 rotations.")
+
+        times = np.asarray(times)
+        if times.ndim != 1:
+            raise ValueError("Expected times to be specified in a 1 "
+                             "dimensional array, got {} "
+                             "dimensions.".format(times.ndim))
+
+        if times.shape[0] != len(rotations):
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of timestamps given, got {} rotations "
+                             "and {} timestamps.".format(
+                                len(rotations), times.shape[0]))
+
+        omega_i = np.asarray(omega_i)
+        if omega_i.shape != (3,):
+            raise ValueError("Expected initial angular velocity `omega_i` to "
+                             "be specified as a 1 dimensional 3-vector, got "
+                             "shape {}.".format(omega_i.shape))
+
+        omega_f = np.asarray(omega_f)
+        if omega_f.shape != (3,):
+            raise ValueError("Expected final angular velocity `omega_f` to "
+                             "be specified as a 1 dimensional 3-vector, got "
+                             "shape {}.".format(omega_f.shape))
+        self.times = times
+        self.timedelta = np.diff(times)
+        if np.any(self.timedelta <= 0):
+            raise ValueError("Times must be in strictly increasing order.")
+
+        # Calculate intermediate angular velocities
+        dt_inv = 1 / self.timedelta
+        rotvecs = (rotations[:-1].inv() * rotations[1:]).as_rotvec()
+        w_int, diff = _intermediate_step(
+            np.zeros((len(rotations) - 2, 3)), rotvecs, dt_inv,
+            omega_i, omega_f)
+
+        # Tolerance for iterative computation. Reference conatins 1e-6 but only
+        # as an estimate for number of iterations. The number of iterations
+        # depends on the 'non-linearity' or lack of 'smoothness' of the curve.
+        # Default of 1e-12 is achieved in ~200 iterations.
+        while diff >= tol:
+            w_int, diff = _intermediate_step(w_int, rotvecs, dt_inv,
+                                             omega_i, omega_f)
+
+        self.omega = np.empty((len(rotations), 3))
+        self.omega[0] = omega_i
+        self.omega[-1] = omega_f
+        self.omega[1:-1] = w_int
+
+        # Formula for theta(t) from eqn 16, coefficients from page 5 of ref
```

Comment:
Let's use [http://scipy.github.io/devdocs/generated/scipy.interpolate.PPoly.html#scipy.interpolate.PPoly](PPoly) to store coefficients and do polynomial evaluations.

Reply:
This will require some re-formulation of the polynomial in the paper, I'll do that.